File: back/src/main/java/br/eng/eliseu/presente/model/dto/ProdutoEstoqueDto.java
package br.eng.eliseu.presente.model.dto;

import br.eng.eliseu.presente.model.StatusEnum;

import java.math.BigDecimal;

public record ProdutoEstoqueDto(
        Long id,
        Long produtoId,

        CorDto cor,
        TamanhoDto tamanho,

        BigDecimal preco,
        BigDecimal quantidade,
        StatusEnum status,
        Long version

) {}

File: front/src/app/shared/model/dto/produto-estoque-dto.ts
import { StatusEnum } from '../enum/status.enum';
import { CorDto } from '@/shared/model/dto/cor-dto';
import { TamanhoDto } from '@/shared/model/dto/tamanho-dto'; // Ajuste o caminho conforme necessário

export class ProdutoEstoqueDto {
    id?: number;
    produtoId?: number;

    cor?: CorDto;
    tamanho?: TamanhoDto;

    preco?: number;
    quantidade?: number;
    status?: StatusEnum;
    version?: number;

}


